backend/dataall/modules/metadata_forms/db/metadata_form_repository.py

```diff
@@ -0,0 +1,35 @@
+from sqlalchemy import or_
+
+from dataall.modules.metadata_forms.db.metadata_form_models import MetadataForm
+from dataall.modules.metadata_forms.db.enums import MetadataFormVisibility
+
+
+class MetadataFormRepository:
+    @staticmethod
+    def create_metadata_form(session, data=None):
+        mf: MetadataForm = MetadataForm(
+            name=data.get('name'),
+            description=data.get('description'),
+            SamlGroupName=data.get('SamlGroupName'),
+            visibility=data.get('visibility'),
+            homeEntity=data.get('homeEntity'),
+        )
+        session.add(mf)
+        session.commit()
+        return mf
+
+    @staticmethod
+    def get_metadata_form(session, uri):
+        return session.query(MetadataForm).get(uri)
+
+    @staticmethod
+    def list_metadata_forms(session, filter=None):
+        query = session.query(MetadataForm)
+        if filter and filter.get('search_input'):
+            query = query.filter(
+                or_(
+                    MetadataForm.name.ilike('%' + filter.get('search_input') + '%'),
+                    MetadataForm.description.ilike('%' + filter.get('search_input') + '%'),
+                )
+            )
+        return query
```

Comment:
can we add an order by condition to ensure stable order for pagination

Reply:
Sure. I will use `MetadataFormSortCriteria` with default option

Reply:
[0c98e09](https://github.com/data-dot-all/dataall/pull/1444/commits/0c98e099ef68ffff6d0109ba7cadd578b9da9c25)